### PR TITLE
feat: refactor product loading with queries

### PR DESCRIPTION
**Pull request by dunqan**
- Introduces base Query and Command functionality
- Refactor product services with Query usages

### CODE REVIEW

File: libs/domain/product/src/services/adapter/default-product-list.adapter.ts [hunk 1]
```diff
@@ -55,13 +55,20 @@ export class DefaultProductListAdapter implements ProductListAdapter {
...
 
   get(qualifier: ProductListQualifier): Observable<ProductList> {
+    const include = [
+      ApiProductModel.Includes.AbstractProducts,
+      ApiProductModel.Includes.ConcreteProducts,
+      ApiProductModel.Includes.ConcreteProductImageSets,
+      ApiProductModel.Includes.ConcreteProductPrices,
+      ApiProductModel.Includes.ConcreteProductAvailabilities,
+      ApiProductModel.Includes.Labels,
+    ].filter((type, index, arr) => arr.indexOf(type) === index);
```
Comment (supproduction) [inline libs/domain/product/src/services/adapter/default-product-list.adapter.ts]: I am wondering, why do we need this filter?

File: libs/platform/core/src/services/query/core/core-query.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    protected destroyNotifier$?: Observable<undefined>
+  ) {}
+
+  protected getKey(qualifier: Qualifier): string {
+    return qualifier
+      ? JSON.stringify(
+          Object.entries(qualifier)
+            .filter((x) => x[1])
+            .sort()
+            .flat()
```
Comment (supproduction) [inline libs/platform/core/src/services/query/core/core-query.ts:55]: mb we should change order?

File: libs/platform/core/src/services/query/core/core-query.ts [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+  }: {
+    data: ValueType;
+    qualifier: Qualifier;
+    optimistic?: boolean;
+  }): void {
+    const subject$ = this.queryCache.get(this.prepareKey(qualifier))!.subject$;
+    subject$.next({
+      error: subject$.value.error,
+      loading: subject$.value.loading,
+      stale: optimistic ? true : subject$.value.stale,
```
Comment (supproduction) [inline libs/platform/core/src/services/query/core/core-query.ts:191]:
```suggestion
      stale: optimistic || subject$.value.stale,
```

File: libs/platform/core/src/services/query/default-query.service.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  QueryService,
+  QUERY_TOKEN,
+} from './query.service';
+
+export class DefaultQueryService
+  extends CoreQueryService
+  implements QueryService, OnDestroy
+{
+  constructor(
+    effects = inject(EFFECT_TOKEN, []),
```
Comment (supproduction) [inline libs/platform/core/src/services/query/default-query.service.ts]: Should we mark it as protected?

File: libs/platform/core/src/services/query/default-query.service.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    protected injector = inject(INJECTOR)
+  ) {
+    super();
+    this.initEffects(effects);
+  }
+
+  protected initEffects(effects: any[]): void {
```
Comment (supproduction) [inline libs/platform/core/src/services/query/default-query.service.ts]: Can omit any?

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/product/src/services/adapter/default-product-list.adapter.ts (hunk 1 above)
Comment (dunqan) [reply libs/domain/product/src/services/adapter/default-product-list.adapter.ts]: thanks!

File: libs/platform/core/src/services/query/default-query.service.ts (hunk 4 above)
Comment (dunqan) [reply libs/platform/core/src/services/query/default-query.service.ts]: 👍

File: libs/platform/core/src/services/query/default-query.service.ts (hunk 5 above)
Comment (dunqan) [reply libs/platform/core/src/services/query/default-query.service.ts]: 👍

File: libs/platform/core/src/services/query/core/core-query.ts (hunk 2 above)
Comment (dunqan) [reply libs/platform/core/src/services/query/core/core-query.ts:55]: no, we want to sort parameters of qualifier before faltting values. Flatting it uset to make key string more compact.

File: libs/platform/core/src/services/query/core/core-query.ts (hunk 3 above)
Comment (dunqan) [reply libs/platform/core/src/services/query/core/core-query.ts:191]: Hmm... 
I like the first one for readibility, and hope that optimizer will optimize it. But not strong opinion :)

**Review by supproduction — APPROVED** [review]

**Pull request merged**